tanner/tests/test_api.py

```diff
@@ -0,0 +1,297 @@
+
+        self.loop.run_until_complete(test())
+        self.handler.apply_filter.assert_has_calls(calls, any_order=True)
+        self.assertEqual(self.expected_content, self.returned_content)
+
+    def test_return_sessions_error(self):
+        self.handler = Api(self.redis_client)
+        self.handler.return_snares = AsyncMock(return_value=["8fa6aa98-4283-4085-bfb9-a1cd3a9e56e4"])
+
+        session = {
+            "sess1": {
+                "attack_types": ["rfi", "lfi"]
+            }
+        }
+        self.handler.return_snare_info = AsyncMock(return_value=session)
+
+        self.filters = {
+            "attacktypes": "lfi"
+        }
+
+        self.expected_content = 'Invalid filter : attacktypes'
+
+        async def test():
+            self.returned_content = await self.handler.return_sessions(self.filters)
+
+        self.loop.run_until_complete(test())
+        self.assertEqual(self.expected_content, self.returned_content)
+
+    def test_apply_filter_user_agent(self):
+        self.handler = Api(self.redis_client)
+        filter_name = 'user_agent'
+        filter_value = 'Mozilla'
+
+        session = {
+            'user_agent': 'Mozilla/5.0'
+        }
+
+        self.returned_content = self.handler.apply_filter(filter_name, filter_value, session)
+        self.assertTrue(self.returned_content)
+
+    def test_apply_filter_user_agent_false(self):
+        self.handler = Api(self.redis_client)
+        filter_name = 'user_agent'
+        filter_value = 'Mozilla Firefox'
+
+        session = {
+            'user_agent': 'Mozilla/5.0'
+        }
+
+        self.returned_content = self.handler.apply_filter(filter_name, filter_value, session)
+        self.assertFalse(self.returned_content)
+
+    def test_apply_filter_possible_owner(self):
+        self.handler = Api(self.redis_client)
+        filter_name = 'possible_owners'
+        filter_value = 'crawler'
+
+        session = {
+            'possible_owners': ['user']
+        }
+
+        self.returned_content = self.handler.apply_filter(filter_name, filter_value, session)
+        self.assertFalse(self.returned_content)
+
+    def test_apply_filter_attack_types(self):
+        self.handler = Api(self.redis_client)
+        filter_name = 'attack_types'
+        filter_value = "xss"
+
+        session = {
+            "attack_types": ["rfi", "xss"]
+        }
+
+        self.returned_content = self.handler.apply_filter(filter_name, filter_value, session)
+        self.assertTrue(self.returned_content)
+
+    def test_apply_filter_attack_types_false(self):
+        self.handler = Api(self.redis_client)
+        filter_name = 'attack_types'
+        filter_value = "lfi"
+
+        session = {
+            "attack_types": ["rfi", "xss"]
+        }
+
+        self.returned_content = self.handler.apply_filter(filter_name, filter_value, session)
+        self.assertFalse(self.returned_content)
+
+    def test_apply_filter_start_time(self):
```

Comment:
I think we can add false results here too.

Reply:
Done.